app/models/hyrax/file_set.rb

```diff
@@ -7,6 +7,8 @@ module Hyrax
   # @see https://wiki.duraspace.org/display/samvera/Hydra%3A%3AWorks+Shared+Modeling
   class FileSet < Hyrax::Resource
     include Hyrax::Schema(:core_metadata)
+    include Hyrax::FileSet::Characterization
```

Comment:
Both @aaron-collier and I **believe** this is the smallest possible set of includes to make sure resource-backed file sets respond to messages that are sent in the course of characterizing them.

`Hyrax::FileSet::Characterization` is needed to set up the characterization proxy, and `Hydra::Works::MimeTypes` is needed to ask a file set what sort of files it wraps (*e.g.*, `#image?`), based on mime types.

Reply:
+1 on including `MimeTypes`, the derivatives job also needs to know that for a resource-backed file set.

Reply:
if these are required `FileSet` interface, let's get them into the `'a Hyrax::FileSet'` shared specs in `lib/hyrax/specs/shared_specs/hydra_works.rb`?

i'm tempted to look for a way around making this the interface, though. so far we've been mostly successful in avoiding putting business logic in the models. while there's definitely space for compromise on that, it seems worth at least poking at other possibilities. any ideas?

Reply:
re: `MimeTypes`,

`Hyrax::FileSet` doesn't (yet) have a `#mime_type` property, so i think the include is injecting a bunch of methods that will currently raise errors?

Reply:
aho noticing that `::FileSet` *also* doesn't have a `mime_type` property and just delegates it to `#characterization_proxy`. it seems like we should follow suit and leave `#mime_type` at the file level?

Reply:
wondering about this as a possible solution to the `#mime_type` issue https://github.com/samvera/hyrax/pull/4215